cfe_internal/enterprise/federation/federation.cf

```diff
@@ -291,29 +331,58 @@ bundle agent entry
   methods:
     config_exists::
       "CFEngine Enterprise Federation Configuration"
+        handle => "config",
         usebundle => config;
-    enterprise_edition.(policy_server|am_policy_hub)::
-      "federation_manage_files"
-        usebundle => federation_manage_files;
     am_superhub|am_feeder::
       "CFEngine Enterprise Federation Transport User"
+        handle => "transport_user",
         usebundle => transport_user;
+    enterprise_edition.(policy_server|am_policy_hub)::
+      "federation_manage_files"
+        handle => "federation_manage_files",
+        usebundle => federation_manage_files;
     enabled::
       "CFEngine Enterprise Federation Postgres Configuration"
+        handle => "postgres_config",
         usebundle => postgres_config;
       "CFEngine Enterprise Federation Feeder Data Export"
         usebundle => exported_data;
       "CFEngine Enterprise Federation Feeder Data Transport"
         usebundle => data_transport;
       "CFEngine Enterprise Federation Feeder Data Import"
         usebundle => imported_data;
+      "Update Status"
+        usebundle => setup_status;
   reports:
     !enterprise_edition::
       "Federated reporting is only available in CFEngine Enterprise.";
     enterprise_edition.!(policy_server|am_policy_hub)::
       "Federated reporting is only available on the policy server / hub.";
 }
 
+bundle agent setup_status
+{
+  classes:
+    "superhub_setup_status_complete"
+      expression => "any",
+      depends_on => {
+                      "config",
+                      "transport_user",
+                      "postgres_config", # We are depending on a deep guard within this bundle
+                      "federation_manage_files",
+      };
+
+  files:
+    superhub_setup_status_complete::
+      "$(cfengine_enterprise_federation:config.federation_dir)/cfapache/setup-status.json"
+        create => "true",
+        template_method => "inline_mustache",
+        edit_template_string => "{{%-top-}}",
+        template_data => '{ "configured": true, "transport_ssh_public_key": "$(with)" }',
+        with => readfile( "$(cfengine_enterprise_federation:transport_user.ssh_pub_key)" ),
```

Comment:
whoa, leveraging `with` too I see. I think that this will need to be placed inside of an if minimum version macro because with isn't valid on 3.10.x, so I would expect this to fail the backward compatibility syntax check.